src/router/index.js

```diff
@@ -179,3 +180,10 @@ export default new Router({
 		},
 	],
 })
+
+router.beforeResolve((from, to, next) => {
+	cancelAll()
+	next()
+})
```

Comment:
We lose aborting request on destroy, useful for components like the `FilePicker`. Wouldn't a mixin be more flexible here ?

Reply:
do you mean putting only the cancelAll() call into a mixin, so views can decide whether they want to use it?

Reply:
And the `abortController` creation, exposing it on `data`.

Something like:


```js
export default {
	name: 'CancelRequestMixin',

	data() {
		return {
			abortController: new AbortController()
		}
	},

	beforeDestroy() {
		this.abortController.abort()
	},

	beforeRouteLeave() {
		this.abortController.abort()
	},
}
```

Reply:
But then how do we get the abort signal to vuex actions? Via params?

Reply:
Yes, the same way you did it on `Tag.vue` for example:


```js
			const files = await getTaggedImages(this.item.injected.id, {
				signal: this.abortController.signal,
			})
```

or

```js
			const files = await getTaggedImages(this.item.injected.id, {
				signal: this.abortSignal,
			})
```